frontend/projects/ui/src/app/services/patch-db/patch-db.factory.ts

```diff
@@ -1,50 +1,57 @@
 import { InjectionToken } from '@angular/core'
-import { exists } from '@start9labs/shared'
-import { filter } from 'rxjs/operators'
-import {
-  Bootstrapper,
-  DBCache,
-  MockSource,
-  PollSource,
-  Source,
-  WebsocketSource,
-} from 'patch-db-client'
-
-import { ConfigService } from '../config.service'
-import { ApiService } from '../api/embassy-api.service'
-import { MockApiService } from '../api/embassy-mock-api.service'
+import { catchError, retry, switchMap, tap, timeout } from 'rxjs/operators'
+import { Bootstrapper, DBCache, Update } from 'patch-db-client'
 import { DataModel } from './data-model'
-import { BehaviorSubject } from 'rxjs'
+import { WebSocketSubjectConfig } from 'rxjs/webSocket'
+import { combineLatest, EMPTY, from, Observable, of } from 'rxjs'
+import { AuthService } from '../auth.service'
+import { ConnectionFailure, ConnectionService } from '../connection.service'
+import { ApiService } from '../api/embassy-api.service'
 
-// [wsSources, pollSources]
-export const PATCH_SOURCE = new InjectionToken<Source<DataModel>[]>('')
-export const PATCH_SOURCE$ = new InjectionToken<
-  BehaviorSubject<Source<DataModel>[]>
->('')
+export const PATCH_SOURCE = new InjectionToken<Observable<Update<DataModel>>>(
+  '',
+)
 export const PATCH_CACHE = new InjectionToken<DBCache<DataModel>>('', {
   factory: () => ({} as any),
 })
 export const BOOTSTRAPPER = new InjectionToken<Bootstrapper<DataModel>>('')
 
-export function mockSourceFactory({
-  mockPatch$,
-}: MockApiService): Source<DataModel>[] {
-  return Array(2).fill(
-    new MockSource<DataModel>(mockPatch$.pipe(filter(exists))),
-  )
-}
+export function sourceFactory(
+  api: ApiService,
+  authService: AuthService,
+  connectionService: ConnectionService,
+): Observable<Update<DataModel>> {
+  const config: WebSocketSubjectConfig<Update<DataModel>> = {
+    url: `/db`,
+    closeObserver: {
+      next: val => {
+        if (val.reason === 'UNAUTHORIZED') authService.setUnverified()
+      },
+    },
+  }
+
+  const websocket$ = api.openPatchWebsocket$(config).pipe(
+    timeout({ first: 21000 }),
+    catchError((e, watch$) => {
+      connectionService.setPatchError(e)
 
-export function realSourceFactory(
-  embassyApi: ApiService,
-  config: ConfigService,
-  { defaultView }: Document,
-): Source<DataModel>[] {
-  const { patchDb } = config
-  const host = defaultView?.location.host
-  const protocol = defaultView?.location.protocol === 'http:' ? 'ws' : 'wss'
+      return of('').pipe(
```

Comment:
What's the purpose of `of('')` and `switchMap`? Why not `from` at the beginning?